Tests/TensorFlowTests/OperatorTests/MathTests.swift

```diff
@@ -199,6 +199,17 @@ final class MathOperatorTests: XCTestCase {
         XCTAssertEqual(0.816997, Double(prediction.scalars[0]), accuracy: 0.0001)
     }
 
+    func testBroadcastedAddGradient() {
+	  func foo(_ x: Tensor<Float>, _ y: Tensor<Float>) -> Tensor<Float> {
+	    return x + y
+	  }
+	  let x = Tensor<Float>(ones: [1, 2, 1, 4])
+	  let y = Tensor<Float>(ones: [4, 1, 3, 1])
+	  let (dx, dy) = gradient(at: x, y, in: foo)
```

Comment:
In this test case, the use of `gradient(at:in:)` is not valid because gradient is only mathematically defined for functions that return a scalar. We can either make `foo(_:_:)` to a `sum()` or use `pullback(at:in:)`.

Reply:
That's what I thought. This was actually the failing case in the other PR and that's why I copied it here but wasn't sure of the semantics of `gradient`. I'll add a call to `sum()` and remove the seed broadcasts then. :)